smarts/sstudio/genhistories.py

```diff
@@ -990,6 +1009,48 @@ def column_val_in_row(self, row, col_name: str) -> Any:
         return row[col_name]
 
 
+def import_dataset(
+    dataset_spec: types.TrafficHistoryDataset, output_path: str, overwrite: bool
+):
+    """called to pre-process (import) a TrafficHistoryDataset for use by SMARTS"""
+    if not dataset_spec.input_path:
+        print(f"skipping placeholder dataset spec '{dataset_spec.name}'.")
+        return
+    output = os.path.join(output_path, f"{dataset_spec.name}.shf")
+    if os.path.exists(output):
+        if not overwrite:
+            print("file already exists at {output}.  skipping...")
+            return
+        os.remove(output)
+    source = dataset_spec.source_type
+    if source.startswith("NGSIM"):
+        dataset = NGSIM(dataset_spec.__dict__, output)
+    elif source == "INTERACTION":
+        dataset = Interaction(dataset_spec.__dict__, output)
+    elif source == "Waymo":
+        dataset = Waymo(dataset_spec.__dict__, output)
+    else:
+        raise ValueError(
+            "unsupported TrafficHistoryDataset type: {dataset_spec.source_type}"
+        )
+    dataset.create_output()
+
+
+def _migrate_deprecated_dataset(dataset_spec: Dict[str, Any]):
+    def remap(old_d: Dict[str, Any], old_key: str, new_d: Dict[str, Any], new_key: str):
+        if new_key not in new_d and old_key in old_d:
+            new_d[new_key] = old_d[old_key]
+
+    remap(dataset_spec, "source", dataset_spec, "source_type")
+    if "map_net" in dataset_spec:
+        map_spec = dataset_spec["map_net"]
+        remap(map_spec, "lane_width", dataset_spec, "map_lane_width")
+        if "max_y" in map_spec or "width" in map_spec or "height" in map_spec:
+            max_x = map_spec.get("width")
+            max_y = map_spec.get("max_y", map_spec.get("height"))
+            dataset_spec["map_bbox"] = BoundingBox((0, 0), (max_x, max_y))
+
```

Comment:
This is to support the deprecated `yaml` files?

Reply:
Yes.  For the time being...  I think in v0.7, we should stop supporting them entirely and then this can be removed.